docker/ubuntu-2004-builder.Dockerfile

```diff
@@ -63,17 +63,18 @@ RUN cd /root \
     && cd /root \
     && rm -rf boost*
 
-# https://github.com/WebAssembly/wasi-sdk/releases/tag/wasi-sdk-20
-ENV WASI_SDK_PREFIX=/usr/lib/llvm-16
+# https://github.com/WebAssembly/wasi-sdk/releases/tag/wasi-sdk-24
+ENV WASI_SDK_PREFIX=/usr/lib/llvm-18
 ENV PATH=${WASI_SDK_PREFIX}/bin:$PATH
-RUN cd ${WASI_SDK_PREFIX}/lib/clang/16/                         \
-    && curl -LO https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-20/libclang_rt.builtins-wasm32-wasi-20.0.tar.gz \
-    && tar xf libclang_rt.builtins-wasm32-wasi-20.0.tar.gz      \
-    && rm libclang_rt.builtins-wasm32-wasi-20.0.tar.gz          \
+RUN cd ${WASI_SDK_PREFIX}/lib/clang/18/                         \
+    && curl -LO https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-24/libclang_rt.builtins-wasm32-wasi-24.0.tar.gz \
+    && mkdir -p lib/wasi                                        \
+    && tar xf libclang_rt.builtins-wasm32-wasi-24.0.tar.gz -C lib/wasi --strip-components=1 \
+    && rm libclang_rt.builtins-wasm32-wasi-24.0.tar.gz          \
     && cd ${WASI_SDK_PREFIX}/share                              \
-    && curl -LO https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-20/wasi-sysroot-20.0.tar.gz \
-    && tar xf wasi-sysroot-20.0.tar.gz                          \
-    && rm wasi-sysroot-20.0.tar.gz
+    && curl -LO https://github.com/WebAssembly/wasi-sdk/releases/download/wasi-sdk-24/wasi-sysroot-24.0.tar.gz \
+    && tar xf wasi-sysroot-24.0.tar.gz                          \
+    && rm wasi-sysroot-24.0.tar.gz
```

Comment:
The old structure was:
`wasi-sysroot-20.0.tar.gz/wasi-sysroot/...`

The new structure is:
`wasi-sysroot-24.0.tar.gz/wasi-sysroot-24.0/...`

We should untar into a directory that is always called `wasi-sysroot` so other configs don't have to change (Like [CMakeLists.txt#L83](https://github.com/gofractally/psibase/blob/main/CMakeLists.txt#L83))

Reply:
Oops. I didn't notice that name change